### ISSUE TITLE
Use Vaadins-Theming logic to change the Theme

### ISSUE THREAD

**Issue by Springrbua** [appreciated/vaadin-app-layout#227]
**Describe the current way to archive your requirement**
If you want to use the `Material`-theme in an application, you have to add the `@Theme(Material.class)` to the top-level Component and you need to call `withTheme(Material.class)` while building the `AppLayout` using the `Builder`.
In version 2.0.7, calling `setTheme` for the `AppLayout` added the stylesheet for the `Theme` to the page.
This has been changed recently and now both theme-stylesheets are imported and the `setTheme` just changes the classname and "activates" one of the stylesheets.

**Describe the solution you'd like**
Instead of defining the `Theme` twice, the `AppLayout` should use vaadins logic to decide which theme to use.
Vaadin seems to load resources which are placed under `frontend/src` from `frontend/theme/lumo` or `frontend/theme/material`, depending on the current theme.
Using this logic, you would only need to rename the `app-layout-lumo` and `app-layout-material` to `app-layout` and move them to the correct folder.

**Additional context**
I already tryed to change the behavior, but unfortunately I failed. My guess is, that the theming-support changed in Vaadin 12 (they included Material in Vaadin 12) and the add-on is using Vaadin 10.
So maybe this change has to wait until the vaadin version is updated?

**Comment by appreciated** [comment]
Hmm I think it is actually a bit more complicated than that. Will postpone to V3.0.

**Comment by Springrbua** [comment]
In the application I am working on, I wanted to create common theme-variants for the `vaadin-button` for both Lumo an Material Theme.
I created a `button-styles.html` file in `theme/lumo` and `theme/material` and added the themes there.
Then I imported it in my top-level component, using `@HtmlImport("src/button-styles.html")` and it imported the correct file from the theme-folder.
Since the difference between `Lumo` and `Material` in your app is just a single `css`-File I thought it might be solved the same way.
